### PR TITLE
[W-8998476] Add preflight task to retrieve available permissions sets

### PR DESCRIPTION
**Pull request by vmarasow**
# Critical Changes

# Changes
We added a new custom task that retrieves the list of available permission sets in the org. 

# Issues Closed
[W-8998476](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008a58bIAA/view)

### CODE REVIEW

File: cumulusci/cumulusci.yml [hunk 1]
```diff
@@ -256,6 +256,10 @@ tasks:
         description: Retrieves a list of the names of any permission sets assigned to the running user.
         class_path: cumulusci.tasks.preflight.permsets.GetPermissionSetAssignments
         group: Salesforce Preflight Checks
+    get_available_permission_set_licenses:
```
Comment (davisagli) [inline cumulusci/cumulusci.yml:259]: @vmarasow  This should be get_available_permission_sets, right?

Also could you please adjust this to NOT rewrite most of the whitespace in cumulusci.yml? (see diff below)

Comment (vmarasow) [reply cumulusci/cumulusci.yml]: Good catch. I got all the whitespace back to the original state.

File: cumulusci/cumulusci.yml [hunk 2]
```diff
@@ -256,6 +256,10 @@ tasks:
         description: Retrieves a list of the names of any permission sets assigned to the running user.
         class_path: cumulusci.tasks.preflight.permsets.GetPermissionSetAssignments
         group: Salesforce Preflight Checks
+    get_available_permission_sets:
+        description: Retrieves a list of the currently available Permission Set definition keys
```
Comment (davisagli) [inline cumulusci/cumulusci.yml:260]:
```suggestion
        description: Retrieves a list of the currently available Permission Sets
```

File: cumulusci/tasks/preflight/licenses.py [hunk 3]
```diff
@@ -23,3 +23,15 @@ def _run_task(self):
         ]
         licenses = "\n".join(self.return_values)
         self.logger.info(f"Found permission set licenses:\n{licenses}")
+
+
+class GetAvailablePermissionSets(BaseSalesforceApiTask):
+    def _run_task(self):
+        self.return_values = [
+            result["Name"]
+            for result in self.sf.query("SELECT Name FROM PermissionSet")[
```
Comment (davisagli) [inline cumulusci/tasks/preflight/licenses.py:32]:
```suggestion
            for result in self.sf.query_all("SELECT Name FROM PermissionSet")[
```

query only returns the first page of results

**Review by davisagli — CHANGES_REQUESTED** [review]
Thanks for the contribution -- I have a few small tweaks to suggest but it looks close.

**Comment by vmarasow** [comment]
Added the suggested changes, please let me know if you can think of anything else

**Comment by davisagli** [comment]
@vmarasow The test also needs a slight change to provide a mock value for query_all rather than query

**Review by davisagli — APPROVED** [review]
Looks good. I'm rerunning the builds because one hit an unrelated error creating a scratch org, but this should be good to merge once the builds are green.

**Review by davisagli — APPROVED** [review]

**Pull request merged**